### PR TITLE
CI--HIP: wget gpg key from https instead of http

### PR DESCRIPTION
**Pull request by etpalmer63**
## Summary
Fixes an issue with HIP CI, now downloads the gpg.key for ROCm sources from https. 

## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

File: .github/workflows/dependencies/dependencies_hip.sh [hunk 1]
```diff
@@ -12,14 +12,15 @@
 
 set -eu -o pipefail
 
+
 # Ref.: https://rocmdocs.amd.com/en/latest/Installation_Guide/Installation-Guide.html#ubuntu
-wget -q -O - http://repo.radeon.com/rocm/rocm.gpg.key \
+wget -q -O - https://repo.radeon.com/rocm/rocm.gpg.key \
   | sudo apt-key add -
 echo 'deb [arch=amd64] http://repo.radeon.com/rocm/apt/debian/ ubuntu main' \
```
Comment (WeiqunZhang) [inline .github/workflows/dependencies/dependencies_hip.sh]: We probably should use https here too?

Comment (etpalmer63) [reply .github/workflows/dependencies/dependencies_hip.sh]: Agreed.

**Review by ax3l — APPROVED** [review]
I don't see issues, but they updated their docs to now use HTTPS, so that's a good idea.
https://rocmdocs.amd.com/en/latest/Installation_Guide/Installation-Guide.html#ubuntu

**Comment by ax3l** [comment]
triggering CI

**Comment by etpalmer63** [comment]
I put it in because of the error message on PR: https://github.com/AMReX-Codes/amrex/pull/2770 But I think its strange its not showing up for others.

**Review by WeiqunZhang — APPROVED** [review]

**Pull request merged**